mgnify_pipelines_toolkit/analysis/shared/dwc_summary_generator.py

```diff
@@ -0,0 +1,235 @@
+    full_res_dict = res_run.json()[0] | res_sample.json()[0]
+
+    fields_to_clean = ["lat", "lon", "depth"]
+
+    for field in fields_to_clean:
+        val = full_res_dict[field]
+        if val == "":
+            full_res_dict[field] = "NA"
+
+    if full_res_dict["collection_date"] == "":
+        full_res_dict["Year"] = "NA"
+        full_res_dict["Month"] = "NA"
+        full_res_dict["Day"] = "NA"
+    else:
+        split_date = full_res_dict["collection_date"].split("-")
+
+        if len(split_date) == 3:
+            full_res_dict["Year"] = split_date[0]
+            full_res_dict["Month"] = split_date[1]
+            full_res_dict["Day"] = split_date[2]
+        elif len(split_date) == 2:
+            full_res_dict["Year"] = split_date[0]
+            full_res_dict["Month"] = split_date[1]
+            full_res_dict["Day"] = "NA"
+        elif len(split_date) == 1:
+            full_res_dict["Year"] = split_date[0]
+            full_res_dict["Month"] = "NA"
+            full_res_dict["Day"] = "NA"
+
+    del full_res_dict["collection_date"]
+
+    res_df = pd.DataFrame(full_res_dict, index=[0])
+    res_df.columns = [
+        "RunID",
+        "SampleID",
+        "StudyID",
+        "decimalLongitude",
+        "depth",
+        "decimalLatitude",
+        "Year",
+        "Month",
+        "Day",
+    ]
+
+    return res_df
+
+
+def get_all_metadata_from_runs(runs):
+
+    run_metadata_dict = defaultdict(dict)
+
+    for run in runs:
+        res_df = get_metadata_from_run_acc(run)
+        if type(res_df) is not bool:
+            run_metadata_dict[run] = res_df
+
+    return run_metadata_dict
+
+
+def cleanup_taxa(df):
+
+    df.pop("Kingdom")
+    cleaned_df = df.rename(columns={"Superkingdom": "Kingdom", "asv": "ASVID"})
+
+    ranks = ["Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species"]
+
+    for rank in ranks:
+        cleaned_df[rank] = cleaned_df[rank].apply(
+            lambda x: x.split("__")[1] if pd.notnull(x) else "NA"
+        )
+
+    for rank in ranks:
+        cleaned_df[rank] = cleaned_df[rank].apply(lambda x: x if x != "" else "NA")
+
+    cleaned_df = cleaned_df[
+        [
+            "ASVID",
+            "StudyID",
+            "SampleID",
+            "RunID",
+            "decimalLongitude",
+            "decimalLatitude",
+            "depth",
+            "Year",
+            "Month",
+            "Day",
+            "Kingdom",
+            "Phylum",
+            "Class",
+            "Order",
+            "Family",
+            "Genus",
+            "Species",
+            "ASVSeq",
+        ]
+    ]
+
+    return cleaned_df
+
+
+def get_asv_dict(runs_df, root_path):
+
+    asv_dict = {}
+    for i in range(0, len(runs_df)):
+        run_acc = runs_df.loc[i, "run"]
+        status = runs_df.loc[i, "status"]
+
+        if status != "all_results":
+            continue
+
+        tax_file = sorted(
+            list(
+                pathlib.Path(f"{root_path}/{run_acc}/asv").glob(
```

Comment:
```suggestion
                (pathlib.Path(root_path) / run_acc / "asv").glob(
```
to avoid `//` mistakes

Reply:
oh this is fancy, I didn't know you could do that!